datalad/export/__init__.py

```diff
@@ -0,0 +1,108 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""
+
+"""
+
+__docformat__ = 'restructuredtext'
+
+import logging
+from glob import glob
+from os.path import join as opj, basename, dirname
+from importlib import import_module
+
+from datalad.support.param import Parameter
+from datalad.support.constraints import EnsureNone
+from datalad.distribution.dataset import EnsureDataset
+from datalad.distribution.dataset import datasetmethod
+from datalad.distribution.dataset import require_dataset
+
+from datalad.interface.base import Interface
+
+lgr = logging.getLogger('datalad.export')
+
+
+def _get_exporter_names():
+    basepath = dirname(__file__)
+    return [basename(e)[:-3]
+            for e in glob(opj(basepath, '*.py'))
+            if not e.endswith('__init__.py')]
```

Comment:
later TODO: could be done through entry points, so others could define additional exporters even outside of datalad codebase.  Look e.g. at example in reprozip and how additional unpackers define themselves via entry points: https://github.com/ViDA-NYU/reprozip/blob/master/reprounzip-docker/setup.py#L16